src/InContextSignup.js

```diff
@@ -32,23 +73,45 @@ export class InContextSignup {
         return isValidTLD() && !isLocalNetwork()
     }
 
-    isAvailable () {
-        const isEnabled = this.device.settings?.featureToggles.emailProtection_incontext_signup
-        const isLoggedIn = this.device.isDeviceSignedIn()
-        return isEnabled && !isLoggedIn && !this.isPermanentlyDismissed() && this.isOnValidDomain() && this.isInstalledRecently
+    isAllowedByDevice () {
+        if (typeof this.isInstalledRecently === 'boolean') {
+            return this.isInstalledRecently
+        } else {
+            // Don't restrict in-context signup based on recent installation
+            // if the device hasn't provided a clear indication
+            return true
+        }
+    }
+
+    /**
+     * @param {import('./Form/matching.js').SupportedSubTypes | "unknown"} [inputType]
+     * @returns {boolean}
+     */
+    isAvailable (inputType = 'emailAddress') {
+        const isEmailInput = inputType === 'emailAddress'
+        const isEmailProtectionEnabled = !!this.device.settings?.featureToggles.emailProtection
+        const isIncontextSignupEnabled = !!this.device.settings?.featureToggles.emailProtection_incontext_signup
+        const isNotAlreadyLoggedIn = !this.device.isDeviceSignedIn()
+        const isNotDismissed = !this.isPermanentlyDismissed()
+        const isOnExpectedPage = this.device.globalConfig.isTopFrame || this.isOnValidDomain()
+        const isAllowedByDevice = this.isAllowedByDevice()
+        return isEmailInput && isEmailProtectionEnabled && isIncontextSignupEnabled && isNotAlreadyLoggedIn && isNotDismissed && isOnExpectedPage && isAllowedByDevice
     }
 
     onIncontextSignup () {
+        this.device.deviceApi.notify(new StartEmailProtectionSignupCall({}))
         this.device.firePixel({pixelName: 'incontext_primary_cta'})
     }
 
-    onIncontextSignupDismissed () {
-        this.device.removeAutofillUIFromPage()
+    onIncontextSignupDismissed (options = { shouldHideTooltip: true }) {
+        if (options.shouldHideTooltip) this.device.removeAutofillUIFromPage()
         this.permanentlyDismissedAt = new Date().getTime()
         this.device.deviceApi.notify(new SetIncontextSignupPermanentlyDismissedAtCall({ value: this.permanentlyDismissedAt }))
         this.device.firePixel({pixelName: 'incontext_dismiss_persisted'})
+        if (options.shouldHideTooltip) this.device.deviceApi.notify(new CloseAutofillParentCall(null))
```

Comment:
Any reason to have this conditional twice?

Reply:
@GioSensation Just that these two control hiding the tooltip. I wasn't sure if the order of calls mattered (remove from page, trigger persisting dismissal data, close parent). If not, I can combine in a single conditional (probably move top line to the bottom?)

Reply:
Yes, I'd move to the bottom. It should make no difference.